Refactor zoom-frame microstate to use spacemacs microstate macro.

ralesi opened the pull request:
Also Modify zoom in and out to keep frame the same size.

syl20bnr commented:
I tested it in 24.3 and it triggers an error `let: Wrong number of arguments: set-frame-size, 4` but the zoom still works. However the frame is still resized.

syl20bnr commented on spacemacs/extensions.el:
```diff
@@ -130,55 +130,43 @@
-        "zf="  'spacemacs/reset-zoom))))
-
-(defun spacemacs/init-emacs-builtin-process-menu ()
-  (evilify process-menu-mode process-menu-mode-map))
```
Should not be deleted.

syl20bnr commented on spacemacs/extensions.el:
```diff
@@ -130,55 +130,43 @@
-         ((> direction 0) (zoom-frm-in)))
-        (when (fboundp 'powerline-reset)
-          (setq-default powerline-height (spacemacs/compute-powerline-height))
-          (powerline-reset))
```
This is important to scale the powerline separators.

ralesi commented:
Think I addressed the issues.  Need to work further on 24.3 compatibility.  Also, fhp is currently off slightly, so frame seems to shrink after repeated use.

syl20bnr commented:
Thank you ! :+1: 
Cherry-picked into develop branch, you can safely delete your branch.
